app/services/uphold/models/transaction.rb

```diff
@@ -28,9 +28,28 @@ def initialize(params = {})
       def all(id:)
         Rails.logger.info("Connection #{@uphold_connection.id} is missing uphold_access_parameters") and return if @uphold_connection.uphold_access_parameters.blank?
 
-        response = get(PATH.expand(id: id), {}, authorization(@uphold_connection))
+        index = 1
+        batch_size = 50
 
-        parse_response(response).map { |a| Transaction.new(a) }
+        transactions = []
+
+        loop do
+          start = (batch_size * index) - batch_size
+          ending = start + batch_size
+
+          response = get(PATH.expand(id: id), {}, authorization(@uphold_connection), { "Range" => "items=#{start}-#{ending}" })
+
+          transactions << parse_response(response).map { |a| Uphold::Models::Transaction.new(a) }
+
+          range = response.headers["content-range"]
+          total_pages = range.split('/').second.to_i
+
+          break if total_pages < ending
```

Comment:
This doesn't seem right at first glance, given that `ending` measures based on entries and `total_pages` is based on `page_index`

Reply:
I agree, just looking at this doesn't make a lot of sense, but breaking out the loop 

```ruby
# 1st loop

start = 0
ending = 50
total_pages = 137
break if total_pages < ending

# 2nd loop
start = 50
ending = 100
total_pages = 137
break if 137 < 100


# 3rd loop
start = 100
ending = 150
total_pages = 137
break if 137 < 150
```

So I think the variables should be renamed maybe to something like `total_items` and then rename `ending` to be  `ending_index`

Reply:
https://github.com/brave-intl/publishers/pull/2330/commits/a6d89d6d16ca96edacdbb20bbfc71019dd3f54bd